Header header

bool left
bool right
